### PR TITLE
Sunset Apollo Server and make GraphQL Helix and Envelop default graphql handler

### PR DESCRIPTION
**Pull request by dthyresson**
Fixes https://github.com/redwoodjs/redwood/issues/3271

* Removes graphql from api package
* Removes logger, webhooks, context from graphql-server package
* Reworks CRWA to use graphql-server imports
* Reworks errors to not be ApolloServer errors
* Reorganizes graphql-server files and test
* Removes the helix e2e tests

### Some background:

>  api is ... auth, functions/webhooks, logging, maybe Global Context? (common api infrasctructurish items ... though are functions/webhooks too Lambda-y?) -- and graphql-server is the schema, parsing, http request pipeline, plugins ... and mapping a resolver to the service.

> An app's api -- if exposing a GraphQL endpoint-- will use both redwoodjs/api and redwoodjs/graphql-server. If the api has some other serverless function that returned a basic RESTful JSON api (json-api.ts) then it might use some other redwood package to implement that data fetching (maybe using same services), then doing that serialization and request/response handling.

### Areas for discussion

* Errors. Should they be more generic RedwoodErrors or tied to Envelop/GraphQL as they are now. We could have generic errors and transform them in the handler and responses (ie via a plugin). For example, Unauthenticated when used in GraphQL could return the Envelop form of it but in a RESTful api return a status 401.
* CRWA is importing from `graphql-server` for the errors and graphql related functions. Keep this? Or import and re-export from api? 
* Global context. Need to test that keeping in api is ok and using w/in graphql-server works as expected. (Update: decide to move to graphql-server)
* Where should beforeResolver live? (Update: moved to graphql-server)
 
### Todos

* revisit README docs in each package
* rework FormError handing in web to not be Apollo Server error specific. Work now, but checks for network errors that no longer will happen with helix/envelop.

### Note

@dac09 and I are refactoring the graphql-server, graphql.ts to extract out each plugin into its own file. We have a different branch for this. But, I did not include that work in this PR.  But we intend to significantly cleanup the createGraphQLHandler.

### CODE REVIEW

**Comment by dac09** [comment]
Headsup @dthyresson ⤵️
Resolved conflicts and trying to fix the TS build error.

**Comment by dac09** [comment]
Still getting the JSON parse error on the redwood query though. On further inspection it looks like its a 401 - I'm seeing this on gitpod.

File: packages/api/tsconfig.json [hunk 1]
```diff
@@ -6,7 +6,7 @@
     "tsBuildInfoFile": "dist/tsconfig.tsbuildinfo",
     "outDir": "dist",
   },
-  "include": ["src/**/*", "src/../package.json"],
+  "include": ["src/**/*"],
```
[the remark this answers is not in the document]
Comment (dac09) [reply packages/api/tsconfig.json:9]: Fixed now.

**Comment by dthyresson** [comment]
> this is so that we bypass tsc for package.json only and the required tsconfig, so don't get weird nesting. <-- what I have done in this branch

I see:

```js
// @NOTE: use require, to avoid messing around with tsconfig and nested output dirs
const packageJson = require('../package.json')
```

Thanks!

**Pull request merged**